packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener.ts

```diff
@@ -142,9 +146,25 @@ export class ApplicationListener extends BaseListener implements IApplicationLis
 
   /**
    * Add one or more certificates to this listener.
+   *
+   * After the first certificate, this creates ApplicationListenerCertificates
+   * resources since cloudformation requires the certificates array on the
+   * listener resource to have a length of 1.
    */
-  public addCertificateArns(_id: string, arns: string[]): void {
-    this.certificateArns.push(...arns);
+  public addCertificateArns(id: string, arns: string[]): void {
+    const additionalCertArns = [...arns];
+
+    if (this.certificateArns.length === 0 && additionalCertArns.length > 0) {
+      const first = additionalCertArns.splice(0, 1)[0];
```

Comment:
using `.shift` feels better but because of the type definition, requires a non-null assertion operator like so:
```
const first = additionalCertArns.shift()!;
```
Otherwise typescript sees the type of `first` being `string | undefined` even though its guarded against by checking that `additionalCertArns.length` is gt 0.

Reply:
Oh yeah, `shift` is totally better, I forgot that existed :)

Reply:
But I'm happy to accept the PR as-is

Reply:
Thanks!